pkg/ast/ast.go

```diff
@@ -1272,10 +1272,24 @@ type ColumnDef struct {
 
 	Null token.Pos // position of "NULL"
 
-	Name    *Ident
-	Type    SchemaType
-	NotNull bool
-	Options *ColumnDefOptions // optional
+	Name          *Ident
+	Type          SchemaType
+	NotNull       bool
+	GeneratedExpr *GeneratedColumnExpr // optional
+	Options       *ColumnDefOptions    // optional
+}
+
+// GeneratedColumnExpr is generated column expression.
+//
+//     AS ({{.Expr | sql}}) STORED
+type GeneratedColumnExpr struct {
```

Comment:
Are `GeneratedColumnExpr.Pos` and `GeneratedColumnExpr.End` missing? They are necessary to implement `Node` interface.

Reply:
Added `Pos`, `End` and `SQL` to `GeneratedColumnExpr`. Also fixed `ColumnDef.End` to take `GeneratedExpr` into account.